### PR TITLE
[TEVA-2550] Remove Validation from Ask For Support Form and Add Callback to Job Application

### PR DESCRIPTION
**Pull request by josephhull676**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-2550

## Comments

- Tried to do the resetting in the form model but couldn't figure out a way to make it work so put it in the job application model. Any suggestions welcome!

### CODE REVIEW

File: app/models/job_application.rb [hunk 1]
```diff
@@ -84,4 +85,8 @@ def fill_in_report
   def reset_equal_opportunities_attributes
     Jobseekers::JobApplication::EqualOpportunitiesForm::ATTRIBUTES.each { |attr| self[attr] = "" }
   end
+
+  def reset_support_needed_details
+    self[:support_needed_details] = "" if support_needed == "no"
```
Comment (cpjmcquillan) [inline app/models/job_application.rb:90]: can you do this?
```ruby
support_needed_details = "" if support_needed == "no"
```

**Review by cpjmcquillan — COMMENTED** [review]
Do you want to add a model spec perhaps?

**Comment by josephhull676** [comment]
> Do you want to add a model spec perhaps?

Yeah was just doing this!

File: app/models/job_application.rb (hunk 1 above)
Comment (cpjmcquillan) [reply app/models/job_application.rb:90]: Shame

File: spec/models/job_application_spec.rb [hunk 2]
```diff
@@ -24,6 +24,18 @@
     end
   end
 
+  context "when setting support needed to 'no'" do
+    subject { create(:job_application) }
```
Comment (cesidio) [inline spec/models/job_application_spec.rb:28]: put some explicit value here for `support_needed_details` so it is clear that it is going to be blanked

File: spec/models/job_application_spec.rb [hunk 3]
```diff
@@ -24,6 +24,18 @@
...
+
+    before do
+      subject.update(support_needed: "no")
+    end
+
+    it "resets support needed details" do
+      expect(subject.support_needed_details).to eq("")
```
Comment (cpjmcquillan) [inline spec/models/job_application_spec.rb:35]: What do you think about if we did something like this instead?

```ruby
before { subject.update(support_needed: "no", support_needed_details: "something secret") }

it "..."
  expect(subject.support_needed).to eq "no"
  expect(subject.support_needed_details).to be_blank
end
```

File: spec/models/job_application_spec.rb (hunk 2 above)
Comment (josephhull676) [reply spec/models/job_application_spec.rb:28]: Sorry yeah that's obvious isn't it lol

**Review by cpjmcquillan — APPROVED** [review]

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**